common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -500,7 +493,7 @@ def import_xblock(self, user_id, course_key, block_type, block_id, fields=None,
                 draft_course = course_key.for_branch(ModuleStoreEnum.BranchName.draft)
                 with self.branch_setting(ModuleStoreEnum.Branch.draft_preferred, draft_course):
                     draft_block = self.import_xblock(user_id, draft_course, block_type, block_id, fields, runtime)
-                    return self.publish(draft_block.location.version_agnostic(), user_id, blacklist=EXCLUDE_ALL, **kwargs)
+                    # Don't publish the draft block here. Instead create the published block below.
```

Comment:
I find this logic hard to understand now (although, to be fair the old logic was confusing too!). So this path is for 'published_only', but it no longer directly publishes the draft but instead it now continues on to do the regular import. It seems that it is now doing more work for 'published_only' than it does for other branch settings, which doesn't seem right intuitively. Could you add some clarifying comments explaining how this logic works in the regular path, and then what is different for ``published_only``.